app/src/hid.c

```diff
@@ -78,16 +79,18 @@ int zmk_hid_implicit_modifiers_release() {
 }
 
 int zmk_hid_keypad_press(zmk_key code) {
-    if (code >= LEFT_CONTROL && code <= RIGHT_GUI) {
-        return zmk_hid_register_mod(code - LEFT_CONTROL);
+    if (code >= HID_EXT_USAGE_ID(HID_USAGE_KEY_KEYBOARD_LEFTCONTROL) &&
+        code <= HID_EXT_USAGE_ID(HID_USAGE_KEY_KEYBOARD_RIGHT_GUI)) {
+        return zmk_hid_register_mod(code - HID_EXT_USAGE_ID(HID_USAGE_KEY_KEYBOARD_LEFTCONTROL));
```

Comment:
`HID_EXT_USAGE_ID` superfluous?

Reply:
No, the code is *just* the usage ID, not the full encoded value, so if you did that, you'd subtract the high bits of the page as well, which is not intended.

Reply:
Oh! I'm sorry, yes, this is.